OSDOCS#7587:Additions and edits to Microshift 4.14 RNs (#64998)
openshift/openshift-docs: microshift_release_notes/microshift-4-14-release-notes.adoc

[id="microshift-4-14-about-this-release"]
== About this release
// TODO: Update with the relevant information closer to release.

> {product-title} was previously introduced as Technology Preview in {product-title} 4.13 and is now Generally Available with {product-title} 4.14. This release uses link:https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md[Kubernetes 1.27] with CRI-O runtime. New features, changes, and known issues that pertain to {product-title} {product-version} are included in this topic.

Author: We also could leave that it was previously introduced as TP in 4.13 since that statement is true, but also add a note that says something like:

"Support for {product-title} 4.13 is planned to end in a future {product-title} release. You should plan to update to the latest version of {product-title}."

//=== Running Applications
//still need to add

//[id="microshift-4-14-lvms-system-requirements"]
//==== LVMS system requirements

[id="microshift-4-14-deprecated-and-removed"]
== Deprecated and removed features

Some features available in previous releases of {product-title} have been deprecated or removed.

Deprecated functionality is still included in {product-title} and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments. For the most recent list of major functionality deprecated and removed within {product-title} {product-version}, refer to the tables below. Additional details for more functionality that has been deprecated and removed are listed after the table.

In the following tables, features are marked with the following statuses:

* _Available_

* _Deprecated_

* _Removed_

.{product-title} deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 | 4.14

|Network configuration flags
|-
|Depricated
|Removed

|CIDR notation
|-
|-
|Removed

|====
[discrete]
[id="microshift-4-14-network-config-flags-removed"]
==== Network configuration flags are removed

> The gateway bridge flag, 'gatewayInterface', and the OVS flag, `disableOVSInit`, in the networking configuration file, `/etc/microshift/ovn.yaml`, are removed with this release. See the xref:../microshift_networking/microshift-networking.adoc#microshift-config-OVN-K_microshift-networking[MicroShift OVN-K configuration] documentation for more information.

Author: @ShaunaDiaz is there a reroute for this link? It is breaking the build and I am not seeing where this file contents went.

Reviewer: [id="microshift-applying-networking-settings"] for the assembly name and [id="microshift-config-OVN-K_{context}"] for the module in the update

Author: Thanks! Just wanted to double check

//[id="microshift-4-14-lvms-system-requirements"]
//==== LVMS system requirements

[id="microshift-4-14-notable-technical-changes"]
== Notable technical changes

{product-title} {product-version} introduces the following notable technical changes.
// Note: use [discrete] for these sub-headings.

[discrete]
[id="microshift-4-14-networking-changes"]
=== Networking changes

Networking updates to {product-title} {product-version} include traffic flow pattern, gateway, and custom configuration changes.

[discrete]
[id="microshift-4-14-traffic-flow-change"]
==== North-south traffic flow changed

The external gateway bridge and the physical device on the host are no longer connected. The north-south traffic between the network service and the OVN external switch flows from the host kernel to {product-title} through the external gateway bridge. See the xref:../microshift_networking/microshift-networking.adoc#microshift-description-connections-network-topology_microshift-networking[MicroShift networking] documentation for more information.

[id="microshift-4-14-deprecated-and-removed"]

> == Deprecated and removed features

Reviewer: @DanielFroehlich Are we following a Deprecation schedule or system with MicroShift's lifecycle? We listed the removed network flags as "deprecated," but I think they are actually removed. Could you clarify if we actually have Deprecated features for 4.14? 
"Deprecated functionality is still included in {product-title} and continues to be supported;"

Some features available in previous releases of {product-title} have been deprecated or removed.

Deprecated functionality is still included in {product-title} and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments. For the most recent list of major functionality deprecated and removed within {product-title} {product-version}, refer to the tables below. Additional details for more functionality that has been deprecated and removed are listed after the table.

In the following tables, features are marked with the following statuses:

* _Available_

* _Deprecated_

* _Removed_

.{product-title} deprecated and removed tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 | 4.14

|Network configuration flags
|-
|-
|Deprecated

|CIDR notation
|-
|-
|Removed

|====
[discrete]
[id="microshift-4-14-network-config-flags-deprecated"]

> ==== Network configuration flags are deprecated

Reviewer: These flags are actually removed from 4.14. That's my mistake. They aren't still supported.

> This section will continue to be updated over time to provide notes on enhancements and bug fixes for future asynchronous errata releases of {product-title} {ocp-version}. Versioned asynchronous releases, for example with the form {product-title} {ocp-version}.z, will be detailed in subsections. In addition, releases in which the errata text cannot fit in the space provided by the advisory will be detailed in subsections that follow.

Reviewer: ```suggestion
This section is updated over time to provide notes on enhancements and bug fixes for future asynchronous errata releases of {product-title} {ocp-version}. Versioned asynchronous releases, for example with the form {product-title} {ocp-version}.z, are detailed in following subsections. In addition, releases in which the errata text cannot fit in the space provided by the advisory are detailed.
```

Author: I wonder if we should actually remove that last sentence. WDYT?

Reviewer: Why not? I am not as familiar with that as you are. Doesn't seem to happen much, and not at all for MicroShift.
